pkg/cluster/delete.go

```diff
@@ -63,3 +90,57 @@ func Delete(
 	}
 	return nil
 }
+
+func DeleteOverrideDir(appIf argoapp.ApplicationServiceClient, kubeClient *kubernetes.Clientset, argocdNs string, name string) error {
+	log := logpkg.GetLogger()
+	app, err := appIf.Get(context.Background(),
+		&apppkg.ApplicationQuery{
+			Name: &name,
+		})
+	if err != nil {
+		return fmt.Errorf("failed to get argocd application: %s", err)
+	}
+	typ := app.Labels["arlon-type"]
+	if typ == "cluster-app" {
+		repoUrl := app.Annotations[baseClusterRepoUrlAnnotation]
+		repoRevision := app.Annotations[baseClusterRepoRevisionAnnotation]
+		repoPath := app.Annotations[baseClusterRepoPathAnnotation] + "/" + name
+		creds, err := argocd.GetRepoCredsFromArgoCd(kubeClient, argocdNs, repoUrl)
+		repo, tmpDir, auth, err := argocd.CloneRepo(creds, repoUrl, repoRevision)
+		if err != nil {
+			return fmt.Errorf("failed to clone repo: %s", err)
+		}
+		wt, err := repo.Worktree()
+		fileInfo, err := wt.Filesystem.Lstat(repoPath)
+		if err == nil {
+			if !fileInfo.IsDir() {
+				return fmt.Errorf("unexpected file type for %s", repoPath)
+			}
+			_, err := wt.Remove(repoPath)
+			if err != nil {
+				return fmt.Errorf("failed to recursively delete cluster directory: %s", err)
+			}
+		}
+
+		commitMsg := "Deleted the files regarding to " + repoPath
+		changed, err := gitutils.CommitDeleteChanges(tmpDir, wt, commitMsg)
+		if err != nil {
+			return fmt.Errorf("failed to commit changes: %s", err)
+		}
+		if !changed {
+			log.Info("no changed files, skipping commit & push")
+			return nil
+		}
+		err = repo.Push(&gogit.PushOptions{
+			RemoteName: gogit.DefaultRemoteName,
+			Auth:       auth,
+			Progress:   nil,
+			CABundle:   nil,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to push to remote repository: %s", err)
+		}
+		log.V(1).Info("successfully pushed working tree", "tmpDir", tmpDir)
+	}
```

Comment:
It's weird to have no else clause here. See my earlier comment about checking the app type.